src/components/generic/InputNumberNoScrollWithUnit/InputNumberNoScrollWithUnit.js

```diff
@@ -28,11 +28,40 @@ const InnerInput = styled(InputNumberNoScroll)`
   padding: ${theme.spacing.xsmall};
 `
 
-const InputNumberNoScrollWithUnit = ({ unit, alignUnitsLeft = false, ...restOfProps }) => {
+const InputNumberNoScrollWithUnit = ({
+  unit = '',
+  alignUnitsLeft = false,
+  value = '',
+  onChange = () => {},
+  name = '',
+  ...restOfProps
+}) => {
+  const [inputValue, setInputValue] = useState(value || '')
+
+  const handleInputChange = (e) => {
```

Comment:
we are intercepting formik's `onChange` handler here to enforce a 2 decimal format